projects/demo-playwright/utils/page-objects/documentation-api-page.po.ts

```diff
@@ -1,21 +1,100 @@
-import {Locator, Page, test} from '@playwright/test';
+import {expect, Locator, Page} from '@playwright/test';
+
+import {tuiHideElement} from '../hide-element';
 
 export class TuiDocumentationApiPagePO {
     readonly apiPageExample: Locator = this.page.locator('#demo-content');
 
-    constructor(private readonly page: Page) {}
+    constructor(protected readonly page: Page) {}
+
+    async waitCompleteLoadingImages(): Promise<void> {
+        const images = await this.apiPageExample.locator('img').all();
+
+        if (images.length) {
+            await Promise.all(
+                images.map(async locator =>
+                    locator.evaluate(
+                        async (image: HTMLImageElement) =>
+                            image.complete ||
+                            new Promise(resolve => {
+                                image.onload = resolve;
+                                image.onerror = resolve;
+                            }),
+                    ),
+                ),
+            );
+        }
+
+        const svgElements = await this.apiPageExample.locator('svg').all();
 
-    setTimeout(n: number): void {
-        test.setTimeout(30_000 * n);
+        if (svgElements.length) {
+            await Promise.all([
+                svgElements.map(async locator =>
+                    locator.evaluate(
+                        (svg: SVGAElement) =>
+                            svg.clientWidth > 0 &&
+                            svg.clientHeight > 0 &&
+                            svg.isConnected,
```

Comment:
Will it really works and wait until all icons loaded? 🤔 
I have doubts that these `localtor.evaluate` will just sometimes return `false` and ... continue execution of script.

Is it expected to use `expect()..toPass()` ?

Reply:
Even **not** loaded svg has not-zero height and width
<img width="1438" alt="Screenshot 2024-01-10 at 10 51 37" src="https://github.com/taiga-family/taiga-ui/assets/35179038/2d818d33-4216-4dbd-8abf-3d1614d74772">

Reply:
\+ in 4.0 all icons will be replaced by `tui-icon`
https://taiga-ui.dev/experimental/icon